patches/tModLoader/Terraria/ModLoader/ModLoader.cs

```diff
@@ -36,7 +36,7 @@ public static class ModLoader
 		public static bool AlphaWelcomed;
 		public static bool ShowFirstLaunchWelcomeMessage;
 
-		public static string versionedName => (ModCompile.DeveloperMode || !BuildInfo.IsRelease) ? BuildInfo.versionedNameDevFriendly : BuildInfo.versionedName;
+		public static string versionedName => (ModCompile.DeveloperMode || !BuildInfo.IsRelease && !BuildInfo.IsBeta) ? BuildInfo.versionedNameDevFriendly : BuildInfo.versionedName;
```

Comment:
We should change this to instead something like the version being `undefined` if it wasn't built from CI. Alternatively, if people want to checkout and build tags, we could use `git name-rev --name-only --tags HEAD` to return the tag of the current commit, which apparently returns `undefined` for non-tagged versions.